dev-libs/rocm-comgr/rocm-comgr-6.1.1.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LLVM_COMPAT=( 17 18 )
+
+inherit cmake llvm-r1 prefix
+
+MY_P=llvm-project-rocm-${PV}
+components=( "amd/comgr" )
+
+DESCRIPTION="Radeon Open Compute Code Object Manager"
+HOMEPAGE="https://github.com/ROCm/ROCm-CompilerSupport"
+SRC_URI="https://github.com/ROCm/llvm-project/archive/rocm-${PV}.tar.gz -> ${MY_P}.tar.gz"
+S="${WORKDIR}/${MY_P}/${components[0]}"
+
+LICENSE="MIT"
+SLOT="0/$(ver_cut 1-2)"
+KEYWORDS="~amd64"
+
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-5.1.3-rocm-path.patch
+	"${FILESDIR}"/0001-Find-CLANG_RESOURCE_DIR-using-clang-print-resource-d.patch
+	"${FILESDIR}"/${PN}-5.7.1-correct-license-install-dir.patch
+	"${FILESDIR}"/${PN}-6.0.0-extend-isa-compatibility-check.patch
+	"${FILESDIR}"/${PN}-6.1.0-remove-incompatible-clang-args.patch
+	"${FILESDIR}"/${PN}-6.1.0-fix-comgr-default-flags.patch
+	"${FILESDIR}"/${PN}-6.1.0-llvm-18-compat.patch
+	"${FILESDIR}"/${PN}-6.1.0-enforce-oop-compiler.patch
+)
+RDEPEND="
+	>=dev-libs/rocm-device-libs-${PV}[${LLVM_USEDEP}]
+	dev-util/hipcc:${SLOT}[${LLVM_USEDEP}]
+	sys-devel/clang-runtime:=
+	$(llvm_gen_dep '
+		sys-devel/clang:${LLVM_SLOT}=
+		sys-devel/lld:${LLVM_SLOT}=
+	')
+"
+DEPEND="${RDEPEND}"
+
+CMAKE_BUILD_TYPE=Release
+
+src_unpack() {
+	if [[ ${PV} == *9999 ]] ; then
+		git-r3_fetch
+		git-r3_checkout '' . '' "${components[@]}"
+	else
+		archive="${MY_P}.tar.gz"
+		ebegin "Unpacking from ${archive}"
+		tar -x -z -o \
+			-f "${DISTDIR}/${archive}" \
+			"${components[@]/#/${MY_P}/}" || die
+		eend ${?}
+	fi
+}
+
+src_prepare() {
+	sed '/sys::path::append(HIPPath/s,"hip","",' -i src/comgr-env.cpp || die
+	sed "/return LLVMPath;/s,LLVMPath,llvm::SmallString<128>(\"$(get_llvm_prefix)\")," \
+		-i src/comgr-env.cpp || die
+
+	eapply $(prefixify_ro "${FILESDIR}"/${PN}-5.0-rocm_path.patch)
+	eapply $(prefixify_ro "${FILESDIR}"/${PN}-6.1.0-fix-tests-rocm-path.patch)
+
+	cmake_src_prepare
+
+	sed "s/@HIP_VERSION@/${PV}/" -i src/comgr-compiler.cpp || die
```

Comment:
Did you miss some patch here which includes @HIP_VERSION@?

Reply:
Oops, feel free to remove this line. I forgot to remove it: it was one of the early ideas on how to fix https://github.com/ROCm/clr/issues/82#issuecomment-2120454786 - I added `--hip-version=@HIP_VERSION@`. Later on I discovered issue with `clang-offload-bundler` invocation, which required forcing out-of-process compilation with hipcc, which adds `--hip-version=...` on its own.